cmd/state-svc/main.go

```diff
@@ -38,28 +38,29 @@ const (
 func main() {
 	var exitCode int
 
-	defer func() {
-		if panics.HandlePanics(recover(), debug.Stack()) {
-			exitCode = 1
-		}
-		if err := events.WaitForEvents(5*time.Second, rollbar.Wait, rollbar.Close, authentication.LegacyClose, logging.Close); err != nil {
-			logging.Warning("Failing to wait for rollbar to close")
-		}
-		os.Exit(exitCode)
-	}()
-
 	cfg, err := config.New()
 	if err != nil {
 		logging.Critical("Could not initialize config: %v", errs.JoinMessage(err))
+		os.Stderr.WriteString(locale.Tr("err_main_config", errs.JoinMessage(err)))
 		exitCode = 1
 		return
 	}
+	logging.CurrentHandler().SetConfig(cfg)
+
 	defer func() {
```

Comment:
This needs to stay on line 41, it can't defer unless the defer was registered first. You could instead register the cfg variable before you define it.

Reply:
Ah right, that makes sense